src/components/veNft/VeNftSetUnclaimedTokensPermissionSection.tsx

```diff
@@ -0,0 +1,52 @@
+import { Trans } from '@lingui/macro'
+import { Button } from 'antd'
+import { NetworkContext } from 'contexts/networkContext'
+import { useUnclaimedTokensPermissionTx } from 'hooks/veNft/transactor/VeNftUnclaimedTokenPermissions'
+import { useContext, useState } from 'react'
+
+const VeNftSetUnclaimedTokensPermissionSection = () => {
+  const { userAddress, onSelectWallet } = useContext(NetworkContext)
+  const [loading, setLoading] = useState(false)
+  const unclaimedTokensPermissionTx = useUnclaimedTokensPermissionTx()
+
+  async function unclaimedTokensPermission() {
+    if (!userAddress && onSelectWallet) {
+      onSelectWallet()
+    }
+
+    setLoading(true)
+
+    const txSuccess = await unclaimedTokensPermissionTx(undefined, {
+      onConfirmed: () => {
+        setLoading(false)
+      },
+    })
+
+    if (!txSuccess) {
+      setLoading(false)
+    }
+  }
+  return (
+    <section>
+      <h3>
+        <Trans>Enable veNFT to Spend Unclaimed Tokens</Trans>
+      </h3>
+      <p>
+        <Trans>
+          This will enable your project's veNFT contract to spend unclaimed
+          tokens in addition to project ERC-20 tokens.
+        </Trans>
+      </p>
+      <Button
+        type="primary"
+        size="small"
+        onClick={unclaimedTokensPermission}
+        loading={loading}
+      >
+        Enable
+      </Button>
+    </section>
+  )
+}
+
+export default VeNftSetUnclaimedTokensPermissionSection
```

Comment:
**question**: Is this the goto way to frame default exports? I have seen a couple of repos like this. Should probably start doing it too

Reply:
i have a snippet in vscode where i type `rafce` and it scaffolds a component like this. i don't know if there's a right or wrong way but that's just my muscle memory for new component files